python/tests/test_pyvw.py

```diff
@@ -259,40 +274,54 @@ def test_multiclass_label():
     assert str(ml) == '2:0.2'
 
 def test_multiclass_label_example():
-    vw_ex = vw(quiet=True)
-    ex = vw_ex.example('1 |a two features |b more features here')
+    n = 4
+    model = pyvw.vw(loss_function='logistic', oaa=n, quiet=True)
+    ex = model.example('1 | a b c d', 2)
     ml2 = pyvw.multiclass_label(ex)
-    assert ml2.label == 1065353216
+    assert ml2.label == 1
     assert ml2.weight == 1.0
     assert ml2.prediction == 0
-    assert str(ml2) == '1065353216'
+    assert str(ml2) == '1'
+
+
+class TestExample:
 
-def test_example():
     vw_ex = vw(quiet=True)
     ex = vw_ex.example('1 |a two features |b more features here')
     ns = pyvw.namespace_id(ex, 1)
-    ns2 = pyvw.namespace_id(ex, 2)
-    assert isinstance(ex.get_ns(1), pyvw.namespace_id)
-    assert isinstance(ex[2], pyvw.example_namespace)
-    assert ex.setup_done is True
-    assert ex.num_features_in(ns) == 3
-    ex.learn()
-    assert ex.setup_done is True
-    ex.unsetup_example() # unsetup an example as it is already setup
-    assert ex.setup_done is False
-    ex.set_label_string('new_label')
-    assert ex.iter_features()
-    assert isinstance(ex.get_label(), pyvw.simple_label)
-    assert ex.get_feature_id(ns, 123) == 123
-    assert ex.get_feature_id(ns, 'a') == 127530
-    ex.push_hashed_feature(ns, 11222)
-    ex.push_features('x', ['a', 'b'])
-    ex.push_features('y', [('c', 1.), 'd'])
-    ex.push_feature(ns, 11000)
-    assert ex.num_features_in(ns) == 5
-    assert ex.num_features_in('x') == ex.num_features_in('y') == 2
-    assert ex.sum_feat_sq(ns) == 5.0
-    ex.push_namespace(ns2)
-    assert ex.pop_namespace()
-    assert ex.iter_features()
-    assert isinstance(ex.get_label(), pyvw.simple_label)
+
+    def test_example_namespace_id(self):
+        ex = self.ex
+        ns = self.ns
+        assert isinstance(ex.get_ns(1), pyvw.namespace_id)
+        assert isinstance(ex[2], pyvw.example_namespace)
+        assert ex.setup_done is True
+        assert ex.num_features_in(ns) == 3
+
+    def test_example_learn(self):
+        ex = self.ex
+        ex.learn()
+        assert ex.setup_done is True
+        ex.unsetup_example() # unsetup an example as it is already setup
+        assert ex.setup_done is False
+
+    def test_example_label(self):
+        ex = self.ex
+        ex.set_label_string('new_label')
+        assert isinstance(ex.get_label(), pyvw.simple_label)
+
+    def test_example_features(self):
+        ex = self.ex
+        ns = self.ns
+        assert ex.get_feature_id(ns, 123) == 123
```

Comment:
Does that feature actually exist in the example?

Reply:
That feature doesn't exists, but I think this was done to avoid raising an error in the case if it doesn't exist, I will change this to assert for an existing feature so it does not misleads someone reading this.